src/components/pages/rules/flyouts/ruleEditor/ruleEditor.js

```diff
@@ -159,10 +160,21 @@ export class RuleEditor extends LinkedComponent {
             },
             error => this.setState({ error, isPending: false, changesApplied: true })
           );
+        logEvent(this.toDiagnosticsModel(requestProps, 'apply_click'));
       }
     }
   }
 
+  toDiagnosticsModel = (rule, eventName) => {
+    const eventProps = {
+      device_group: rule.groupId,
```

Comment:
Why are we introducing underscores?
Is this the "request" model that is sent to the server?  If so, it should be named toDiagnosticsRequestModel.

Reply:
This is neither a request model nor UI vars, This is how we need this data to be saved.